src/citations.tsx

```diff
@@ -1,20 +1,101 @@
+import type { CitationRenderer } from 'citation-js-utils';
+import { InlineCite } from 'citation-js-utils';
 import type { Plugin } from 'unified';
-import type { Root } from 'myst-spec';
-import type { GenericNode } from 'myst-common';
+import type { StaticPhrasingContent, Parent, Root } from 'myst-spec';
+import type { References } from 'myst-common';
 import { selectAll } from 'unist-util-select';
+import type { Cite, CiteKind, CiteGroup } from 'myst-spec-ext';
 
-/**
- * Add fake children to the citations
- */
-export async function addCiteChildrenTransform(tree: Root): Promise<void> {
-  const links = selectAll('cite', tree) as GenericNode[];
-  links.forEach(async cite => {
-    if (cite.children && cite.children.length > 0) return;
-    cite.error = true;
-    cite.children = [{ type: 'text', value: cite.label }];
+function pushCite(
+  references: Pick<References, 'cite'>,
+  citeRenderer: CitationRenderer,
+  label: string
+) {
+  if (!references.cite) {
+    references.cite = { order: [], data: {} };
+  }
+  if (!references.cite?.data[label]) {
+    references.cite.order.push(label);
+  }
+  references.cite.data[label] = {
+    // TODO: this number isn't right? Should be the last time it was seen, not the current size.
+    number: references.cite.order.length,
+    doi: citeRenderer[label]?.getDOI(),
+    html: citeRenderer[label]?.render()
+  };
+}
+
+export function combineCitationRenderers(renderers: CitationRenderer[]) {
```

Comment:
A bunch of this is copied from the CLI, we need to create a package, share this functionality and delete it here!